rexx/RexxParser.g4

```diff
@@ -226,7 +226,20 @@ comparison                  :   concatenation ( comparison_operator concatenatio
                             |   CMPS_NM
                             |   CMPS_NL
                             ;
-concatenation               :   addition ( CONCAT? addition )* ;
+concatenation               :   addition (concatenation_op addition)* ;
+ concatenation_op           :   {
+                                   (getTokenStream().get(getCurrentToken().getTokenIndex()-1).getChannel() == RexxLexer.WHITESPACE_CHANNEL)
```

Comment:
Why did you decide to use another channel here? It's possible to do it using just token type.

```g4
getTokenStream().get(getCurrentToken().getTokenIndex()-1).getTokenType()
```

Reply:
It was an idea I picked up from two [StackOverflow answers that seemed related to my problem: [Allow Whitespace sections ANTLR4]( https://stackoverflow.com/a/29115489/241753) and [Antlr4: How can I both hide and use Tokens in a grammar](https://stackoverflow.com/a/41679234/241753).  I simplified the idea a bit, but apparently not enough.  I guess I got fixated on the idea of a separate channel, and missed the next simplification step of just matching the token itself, not the channel it's on.

Reply:
To answer the other question you asked by email, "_How does this [Move whitespace from the HIDDEN channel to a new WHITESPACES channel] help to detect in parser? Tokens can be extracted from WHITESPACES channel as well as from HIDDEN channel._", the HIDDEN channel carried other tokens as well.  But as you pointed out, I may not have needed a channel change to do this at all.

Reply:
@KvanTTT Does this change resolve your concern?  If so, I'll submit the PR.  I also added a few more test cases, which all pass.

https://github.com/antlr/grammars-v4/compare/master...RossPatterson:grammars-v4:issue_2740_concat_review?expand=1

Reply:
It looks like the link is incorrect. Could you please another one?

Reply:
Hmm ... link works for me.  Maybe GitHub won't let others view it.  Here's the diff from `git log -c`:

```diff
commit 37d69f57ee646bdc0672f46f0ec3e8660727ef1f
Author: Ross Patterson <ross.patterson@gmail.com>
Date:   Mon Aug 15 16:23:43 2022 -0400

    [rexx] Address simplicification comments in PR 2746

diff --git a/rexx/RexxLexer.g4 b/rexx/RexxLexer.g4
index 264b89eb..e509bdbf 100644
--- a/rexx/RexxLexer.g4
+++ b/rexx/RexxLexer.g4
@@ -1,5 +1,4 @@
 lexer grammar RexxLexer;
-channels { WHITESPACE_CHANNEL }
 
 // Main rules
 // %INCLUDE statement
@@ -9,7 +8,7 @@ LINE_COMMENT                    :   Line_Comment_
                                 ->  channel(HIDDEN);
 BLOCK_COMMENT                   :   Block_Comment_
                                 ->  channel(HIDDEN);
-WHITESPACES                     :   Whitespaces_            -> channel(WHITESPACE_CHANNEL);
+WHITESPACES                     :   Whitespaces_            -> channel(HIDDEN);
 CONTINUATION                    :   Continue_               -> channel(HIDDEN);
 
 // Keywords
diff --git a/rexx/RexxParser.g4 b/rexx/RexxParser.g4
index 1689b3a7..b6ca254c 100644
--- a/rexx/RexxParser.g4
+++ b/rexx/RexxParser.g4
@@ -228,12 +228,12 @@ comparison                  :   concatenation ( comparison_operator concatenatio
                             ;
 concatenation               :   addition (concatenation_op addition)* ;
  concatenation_op           :   {
-                                   (getTokenStream().get(getCurrentToken().getTokenIndex()-1).getChannel() == RexxLexer.WHITESPACE_CHANNEL)
+                                   (getTokenStream().get(getCurrentToken().getTokenIndex()-1).getType() == RexxLexer.WHITESPACES)
                                 }? blank_concatenation_op // If previous token is whitespace, this is blank-concatenation.
                             |   normal_concatenation_op
                             ;
   normal_concatenation_op   :   {
-                                   (getTokenStream().get(getCurrentToken().getTokenIndex()-1).getChannel() != RexxLexer.WHITESPACE_CHANNEL)
+                                   (getTokenStream().get(getCurrentToken().getTokenIndex()-1).getType() != RexxLexer.WHITESPACES)
                                 }? // If previous token is not whitespace, this is abuttal-concatenation.
                                 // Note: no token or rule to match here, just the predicate.
                             |   CONCAT
diff --git a/rexx/examples/test_concatenation.rex b/rexx/examples/test_concatenation.rex
index f2274278..e11bc229 100644
--- a/rexx/examples/test_concatenation.rex
+++ b/rexx/examples/test_concatenation.rex
@@ -2,12 +2,21 @@
 
 /* Explicit concatenation: */
 say "Now is the winter " || "of our discontent"
+say "Now is the winter " || /*comment*/ "of our discontent"
+say "Now is the winter " || ,
+"of our discontent"
 
 /* Abuttal concatenation: */
 say "Made glorious "Summer()
+say "Made glorious "/*comment*/Summer()
+say "Made glorious ",
+Summer()
 
 /* Blank concatenation: */
 say "By this sun" "of York"
+say "By this sun"/*comment*/ "of York"
+say "By this sun" ,
+ "of York"
 
 exit
 


```

Reply:
Great! It looks better!

Reply:
PR #2759 submitted.